### PR TITLE
zib-Wound and zib-PressureUlcer

### PR DESCRIPTION
**Pull request by furoreSebastiaan**
Please review: Heads-up: name of branch is misleading, zib-wound is also present.

### CODE REVIEW

File: Profiles - ZIB 2017/woundObservable.xml [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+        <map value="NL-CM:19.1.9" />
+      </mapping>
+      <mapping>
+        <identity value="hcim-Wound-v3.1-2017EN" />
+        <map value="NL-CM:19.2.12" />
+      </mapping>
+    </element>
+    <element id="Observation.component:WoundWidth">
+      <path value="Observation.component" />
+      <sliceName value="WoundWidth" />
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/woundObservable.xml]: `<max value="1" />`

File: Profiles - ZIB 2017/woundObservable.xml [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+        <map value="NL-CM:19.2.13" />
+      </mapping>
+      <mapping>
+        <identity value="hcim-PressureUlcer-v3.2-2017EN" />
+        <map value="NL-CM:19.1.10" />
+      </mapping>
+    </element>
+    <element id="Observation.component:WoundDepth">
+      <path value="Observation.component" />
+      <sliceName value="WoundDepth" />
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/woundObservable.xml]: `<max value="1" />`

File: Profiles - ZIB 2017/woundObservable.xml [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+      </mapping>
+      <mapping>
+        <identity value="hcim-Wound-v3.1-2017EN" />
+        <map value="NL-CM:19.2.14" />
+      </mapping>
+    </element>
+    <element id="Observation.component:WoundTissue">
+      <path value="Observation.component" />
+      <sliceName value="WoundTissue" />
+    </element>
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/woundObservable.xml]: `<max value="1" />`

File: Profiles - ZIB 2017/woundObservable.xml [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+      <min value="1" />
+    </element>
+    <element id="Observation.component:WoundTissue.value[x]:WCSCodelistValue.coding.code">
+      <path value="Observation.component.value[x].coding.code" />
+      <min value="1" />
+    </element>
+    <element id="Observation.component:WoundInfection">
+      <path value="Observation.component" />
+      <sliceName value="WoundInfection" />
+    </element>
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/woundObservable.xml]: `<max value="1" />`

File: Profiles - ZIB 2017/woundObservable.xml [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+      <path value="Observation.component.code.coding.code" />
+      <fixedCode value="405009004" />
+    </element>
+    <element id="Observation.component:WoundInfection.value[x]">
+      <path value="Observation.component.value[x]" />
+      <max value="0" />
+    </element>
+    <element id="Observation.component:WoundMoisture">
+      <path value="Observation.component" />
+      <sliceName value="WoundMoisture" />
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/woundObservable.xml]: `<max value="1" />`

File: Profiles - ZIB 2017/woundObservable.xml [hunk 6]
```diff
@@ -0,0 +1,529 @@
...
+        </valueSetReference>
+      </binding>
+      <mapping>
+        <identity value="hcim-Wound-v3.1-2017EN" />
+        <map value="NL-CM:19.2.6" />
+      </mapping>
+    </element>
+    <element id="Observation.component:WoundEdge">
+      <path value="Observation.component" />
+      <sliceName value="WoundEdge" />
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/woundObservable.xml]: `<max value="1" />`

File: Profiles - ZIB 2017/woundObservable.xml [hunk 7]
```diff
@@ -0,0 +1,529 @@
...
+          <path value="code" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+      <min value="1" />
+    </element>
+    <element id="Observation.code.coding:WoundObservableCode">
+      <path value="Observation.code.coding" />
+      <sliceName value="WoundObservableCode" />
+      <min value="1" />
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/woundObservable.xml]: `<max value="1" />`

File: Profiles - ZIB 2017/zib-Wound.xml [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+      <path value="Condition.onsetDateTime" />
+      <sliceName value="onsetDateTime" />
+      <mapping>
+        <identity value="hcim-Wound-v3.2-2017EN" />
+        <map value="NL-CM:19.2.2" />
+      </mapping>
+    </element>
+    <element id="Condition.note">
+      <path value="Condition.note" />
+      <mapping>
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/zib-Wound.xml]: Maybe put the mapping on Condition.note.text?

File: Profiles - ZIB 2017/Extensions and base profiles/extension-dateOfLastDressingChange-stu3.xml [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    </element>
+    <element id="Extension.url">
+      <path value="Extension.url" />
+      <fixedUri value="http://nictiz.nl/fhir/StructureDefinition/DateOfLastDressingChange" />
+    </element>
+    <element id="Extension.value[x]:valueDate">
+      <path value="Extension.valueDate" />
+      <sliceName value="valueDate" />
+      <type>
+        <code value="date" />
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/Extensions and base profiles/extension-dateOfLastDressingChange-stu3.xml]: HCIM TS maps to FHIR dateTime

File: Profiles - ZIB 2017/Extensions and base profiles/extension-observation-reasonreference-stu3.xml [hunk 10]
```diff
@@ -0,0 +1,227 @@
+<StructureDefinition xmlns="http://hl7.org/fhir">
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/Extensions and base profiles/extension-observation-reasonreference-stu3.xml]: No need to include this file!

File: Profiles - ZIB 2017/Extensions and base profiles/extension-valueBoolean.xml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  <publisher value="Nictiz" />
+  <contact>
+    <name value="Nictiz" />
+    <telecom>
+      <system value="email" />
+      <value value="info@nictiz.nl" />
+      <use value="work" />
+    </telecom>
+  </contact>
+  <description value="A STU3 workaround for Observation.component. The specs ar eincomplete, valueBoolean is not added as a possibility in STU3. It is in R4. With this extension valueBoolean is made possible " />
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/Extensions and base profiles/extension-valueBoolean.xml]: Actually, the STU3 spec defines another workaround: " For boolean values use valueCodeableConcept and select codes from <http://hl7.org/fhir/ValueSet/v2-0136> (these "yes/no" concepts can be mapped to the display name "true/false" or other mutually exclusive terms that may be needed")."

File: Profiles - ZIB 2017/zib-Wound-MedicalDevice.xml [hunk 12]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/zib-Wound-MedicalDevice.xml]: Best practice to add a "purpose" field describing how this profile relates to the HCIM

File: Profiles - ZIB 2017/zib-Wound-MedicalDeviceProduct.xml [hunk 13]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/zib-Wound-MedicalDeviceProduct.xml]: Best practice to add a "purpose" field describing how this profile relates to the HCIM

File: Profiles - ZIB 2017/zib-PressureUlcer.xml [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+        </valueSetReference>
+      </binding>
+      <mapping>
+        <identity value="hcim-PressureUlcer-v3.2-2017EN" />
+        <map value="NL-CM:19.1.2" />
+      </mapping>
+    </element>
+    <element id="Condition.note">
+      <path value="Condition.note" />
+      <mapping>
```
Comment (pieter-edelman-nictiz) [inline Profiles - ZIB 2017/zib-PressureUlcer.xml]: Maybe move the mapping to Condition.note.text?

File: Examples/zib-PressureUlcer-woundObservable - example.xml [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+            <coding>
+                <system value="http://snomed.info/sct"/>
+                <code value="401239006"/>
+                <display value="breedte van verwonding" />
+            </coding>
+        </code>
+        <valueQuantity>
+            <value value="2"/>
+            <unit value="cm"/>
+            <system value="http://unitsofmeasure.org"/>
```
Comment (pieter-edelman-nictiz) [inline Examples/zib-PressureUlcer-woundObservable - example.xml]: Also include a code if you include a system

File: Examples/zib-Wound-woundObservable - example.xml [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+            <coding>
+                <system value="http://snomed.info/sct"/>
+                <code value="401238003"/>
+                <display value="lengte van verwonding" />
+            </coding>
+        </code>
+        <valueQuantity>
+            <value value="5"/>
+            <unit value="cm"/>
+            <system value="http://unitsofmeasure.org"/>
```
Comment (pieter-edelman-nictiz) [inline Examples/zib-Wound-woundObservable - example.xml]: Include code if you include system

File: Examples/zib-Wound-woundObservable - example.xml [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+            <coding>
+                <system value="http://snomed.info/sct"/>
+                <code value="401239006"/>
+                <display value="breedte van verwonding" />
+            </coding>
+        </code>
+        <valueQuantity>
+            <value value="2"/>
+            <unit value="cm"/>
+            <system value="http://unitsofmeasure.org"/>
```
Comment (pieter-edelman-nictiz) [inline Examples/zib-Wound-woundObservable - example.xml]: Include code if you include system

File: Examples/zib-Wound-woundObservable - example.xml [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+            <coding>
+                <system value="http://snomed.info/sct"/>
+                <code value="425094009"/>
+                <display value="diepte van verwonding" />
+            </coding>
+        </code>
+        <valueQuantity>
+            <value value="1"/>
+            <unit value="cm"/>
+            <system value="http://unitsofmeasure.org"/>
```
Comment (pieter-edelman-nictiz) [inline Examples/zib-Wound-woundObservable - example.xml]: Include code if you include system

File: Profiles - ZIB 2017/zib-Wound.xml (hunk 8 above)
Comment (furoreSebastiaan) [reply Profiles - ZIB 2017/zib-Wound.xml]: okay

File: Profiles - ZIB 2017/Extensions and base profiles/extension-observation-reasonreference-stu3.xml (hunk 10 above)
Comment (furoreSebastiaan) [reply Profiles - ZIB 2017/Extensions and base profiles/extension-observation-reasonreference-stu3.xml]: deleted it.

File: Profiles - ZIB 2017/Extensions and base profiles/extension-valueBoolean.xml (hunk 11 above)
Comment (furoreSebastiaan) [reply Profiles - ZIB 2017/Extensions and base profiles/extension-valueBoolean.xml]: Used spec workaround

File: Profiles - ZIB 2017/zib-Wound-MedicalDeviceProduct.xml (hunk 13 above)
Comment (furoreSebastiaan) [reply Profiles - ZIB 2017/zib-Wound-MedicalDeviceProduct.xml]: done

File: Examples/zib-Wound-woundObservable - example.xml (hunk 16 above)
Comment (furoreSebastiaan) [reply Examples/zib-Wound-woundObservable - example.xml]: OK

File: Examples/zib-Wound-woundObservable - example.xml (hunk 17 above)
Comment (furoreSebastiaan) [reply Examples/zib-Wound-woundObservable - example.xml]: OK

File: Examples/zib-Wound-woundObservable - example.xml (hunk 18 above)
Comment (furoreSebastiaan) [reply Examples/zib-Wound-woundObservable - example.xml]: OK

File: Profiles - ZIB 2017/Extensions and base profiles/extension-dateOfLastDressingChange-stu3.xml (hunk 9 above)
Comment (furoreSebastiaan) [reply Profiles - ZIB 2017/Extensions and base profiles/extension-dateOfLastDressingChange-stu3.xml]: DONE

File: Profiles - ZIB 2017/woundObservable.xml (hunk 1 above)
Comment (furoreSebastiaan) [reply Profiles - ZIB 2017/woundObservable.xml]: Done

**Comment by ArdonToonstra** [comment]
Performed various fixes and improvements to Wound and PressureUlcer. Ready to merge. Only extension woundImage needs a description and purposes.

**Comment by ArdonToonstra** [comment]
And there are MISSING codes. Please keep them in sight.

**Pull request merged**